edx_exams/apps/core/management/commands/bulk_add_course_staff.py

```diff
@@ -63,32 +63,26 @@ def add_course_staff_from_csv(self, csv_file, batch_size, batch_delay):
         Add the given set of course staff provided in csv
         """
         reader = list(csv.DictReader(csv_file))
-        users_to_create = []
-        users_existing = {u.username for u in User.objects.filter(username__in=[r.get('username') for r in reader])}
-        for row in reader:
-            if row.get('username') not in users_existing:
-                users_to_create.append(row)
-                users_existing.add(row.get('username'))
 
         # bulk create users
-        for i in range(0, len(users_to_create), batch_size):
+        for i in range(0, len(reader), batch_size):
             User.objects.bulk_create(
-                User(
-                    username=user.get('username'),
-                    email=user.get('email'),
-                )
-                for user in users_to_create[i:i + batch_size]
+                (User(
+                    username=row.get('username'),
+                    email=row.get('email'),
+                ) for row in reader[i:i + batch_size]),
```

Comment:
Will this line generate indexing errors? For example, the `i+batch_size > len(reader)` condition?

Reply:
Good call, let me add testing/handling for that.

Reply:
So I did add tests to test that this still works. Also, I believe that this works because of this line: `for i in range(0, len(reader), batch_size)`. Since this is incremented by batch_size, the line `for row in reader[i:i + batch_size])` won't actually get to the index error. I actually originally repurposed this from some batching examples in the platform, so there are other examples of this pattern being used. See below:

- https://github.com/search?q=repo%3Aopenedx%2Fedx-platform%20%2B%20batch_size&type=code
- https://github.com/openedx/edx-platform/blob/8d0ada54c428b0c546119338d3661980ed4d23d2/scripts/user_retirement/utils/utils.py#L22
- https://github.com/openedx/edx-platform/blob/8d0ada54c428b0c546119338d3661980ed4d23d2/openedx/core/djangoapps/notifications/utils.py#L56

Reply:
okay this was throwing me off too. `reader[i:i + batch_size]` is actually slicing a subset `reader` into a new list. If the `k` in list[i:k] is outside the bounds it won't matter it only pulls the subset of indexes that actually exist up to `k`.